flox/xarray.py

```diff
@@ -326,21 +328,23 @@ def xarray_reduce(
             raise NotImplementedError(
                 "flox does not support binning into an integer number of bins yet."
             )
+
+        expect1: T_ExpectOpt
         if expect is None:
             if isbin_:
                 raise ValueError(
                     f"Please provided bin edges for group variable {idx} "
                     f"named {group_name} in expected_groups."
                 )
-            expect_ = _get_expected_groups(b_.data, sort=sort)
+            expect1 = _get_expected_groups(b_.data, sort=sort)
         else:
-            expect_ = expect
-        expect_index = _convert_expected_groups_to_index((expect_,), (isbin_,), sort=sort)[0]
+            expect1 = expect
+        expect_index = _convert_expected_groups_to_index((expect1,), (isbin_,), sort=sort)[0]
```

Comment:
Prefer `expect_` to `expect1`

Reply:
I found using numbers were much easier to read when you had multiple `expect` with varying types:
`expect1` similar readabilty to `expect_`
`expect2` much easier to read than `expect__`.